students/arkadiusz_kasprzyk/lesson_05_lists/charachter_picture_grid.py

```diff
@@ -0,0 +1,17 @@
+"""
+Prints the transposition of the given grid:
+"""
+grid = [['.', '.', '.', '.', '.', '.'],
+        ['.', 'O', 'O', '.', '.', '.'],
+        ['O', 'O', 'O', 'O', '.', '.'],
+        ['O', 'O', 'O', 'O', 'O', '.'],
+        ['.', 'O', 'O', 'O', 'O', 'O'],
+        ['O', 'O', 'O', 'O', 'O', '.'],
+        ['O', 'O', 'O', 'O', '.', '.'],
+        ['.', 'O', 'O', '.', '.', '.'],
+        ['.', '.', '.', '.', '.', '.']]
+
+for y in range(len(grid[0])):
+    for x in range(len(grid)):
+        print(grid[x][y], end="")
+    print('')
```

Comment:
`print(''.join(''.join(zip(*grid))))`? Just kiddin'

Reply:
I tried it and it doesn't work...
The most similar I can imagine is:
     for k in map(lambda t: print(''.join(list(t))) , zip(*grid)): k
what is rather ugly and few levels above...

Reply:
I was close... after some testing: `print('\n'.join(map(''.join, zip(*grid))))`

Your version works for me too, the dots are printed.

Reply:
That's funny... 
Both versions works the same: everything is printed when you run the whole script (Ctrl+Shift+F10) and 3 dots disappear if you run interactively via Python Console (Ctrl+Alt+E). 
So this is a console problem.
Thanks for the tip; I didn't know that `lambda` keyword may be ommited, and that .join() works on map :) (hence on any iterable i guess).

Reply:
The only times you need to unpack iterables is debugging and preparing the end result `list()` or `tuple()` works great for the latter.

Reply:
Strange. The console might interpret that as continuation line from IPython.

Reply:
Certainly, but how to change this behaviour? (probably very hard to find it, so it doesn't matter, but it worth knowing that such things happen)